### PR TITLE
fix: update attribute_keys to include type

### LINKED ISSUE #98: Should attribute_keys in stream configuration support include/exclude syntax?
The specification states:

> attribute_keys: This is, at a minimum, an allow-list of attribute keys for measurements captured in the metric stream. The allow-list contains attribute keys that identify the attributes that MUST be kept, and all other attributes MUST be ignored.

But looking at the go implementation, an attribute filter can be specified either to include or exclude based on the attribute keys:

https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/metric/instrument.go#L146

I checked the python implementations, and it looks like that implementation only supports including attributes based on the keys.

### PR DESCRIPTION
**Pull request by codeboten**
This updates the stream's `attribute_keys` configuration to an include/exclude style object that currently only supports `included`. This is in hope to support `excluded` in the future once the specification allows it.

Fixes #98

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * Adding initial instrumentation configuration schema
 * Split MetricExporter into PullMetricExporter and PushMetricExporter and ensure only PushMetricExporters can be associated with PeriodicMetricReader [#110](https://github.com/open-telemetry/opentelemetry-configuration/pull/110)
+* Update `attribute_keys` from array to to Include type
```
Comment (codeboten) [inline CHANGELOG.md:7]:
```suggestion
* Update `attribute_keys` from array to to Include type [#111](https://github.com/open-telemetry/opentelemetry-configuration/pull/111)
```

File: examples/kitchen-sink.yaml [hunk 2]
```diff
@@ -249,8 +249,10 @@ meter_provider:
             record_min_max: true
         # Configure attribute keys retained in the resulting stream(s).
         attribute_keys:
-          - key1
-          - key2
+          # Configure list of attribute key patterns attribute keys retained in the resulting stream(s).
```
Comment (MrAlias) [inline examples/kitchen-sink.yaml:252]:
```suggestion
          # Configure list of attribute key patterns that are retained in the resulting stream(s).
          # All other attributes not matching will be dropped.
```

**Review by MrAlias — APPROVED** [review]

File: examples/kitchen-sink.yaml [hunk 3]
```diff
@@ -249,8 +249,11 @@ meter_provider:
             record_min_max: true
         # Configure attribute keys retained in the resulting stream(s).
         attribute_keys:
-          - key1
-          - key2
+          # Configure list of attribute key patterns that are retained in the resulting stream(s).
+          # All other attributes not matching will be dropped.
+          included:
```
Comment (jack-berg) [inline examples/kitchen-sink.yaml:254]: The expansion from absolute values to patterns extends beyond the language of the spec. My understanding is that this adjustment is to help improve ergonomics for when the spec also includes the ability to exclude values. Can we start with absolute keys and evolve to allow patterns if the spec adds support?

File: examples/kitchen-sink.yaml [hunk 4]
```diff
@@ -249,8 +249,11 @@ meter_provider:
             record_min_max: true
         # Configure attribute keys retained in the resulting stream(s).
         attribute_keys:
-          - key1
-          - key2
+          # Configure list of attribute key patterns that are retained in the resulting stream(s).
```
Comment (codeboten) [inline examples/kitchen-sink.yaml:252]:
```suggestion
          # Configure list of attribute keys that are retained in the resulting stream(s).
```

File: examples/kitchen-sink.yaml (hunk 3 above)
Comment (codeboten) [reply examples/kitchen-sink.yaml:254]: Done

**Review by jack-berg — APPROVED** [review]

**Pull request merged**